code/iaas/allocator/src/main/java/io/cattle/platform/allocator/dao/AllocatorDao.java

```diff
@@ -62,4 +62,6 @@
     String getAllocatedHostUuid(Volume volume);
 
     Iterator<AllocationCandidate> iteratorHosts(List<String> orderedHostUuids, List<Long> volumes, QueryOptions options);
+
+    void updateInstanceForNewPorts(List<Map<String, Object>> dataList, List<Port> newPorts);
```

Comment:
Usually it is not a very good idea to pass a generic field to the method call. Can you see if we can make the "dataList" type more concrete + the name should reflect the meaning, dataList as a name is too generic as well

Reply:
Also are we just updating ports in this method, or instance as well? If just ports, the function name should reflect that

Reply:
The name can be changed such as "allocatedDataList", since this dataList is like: [{allocatedIPs=[{allocatedIP=192.168.60.60, privatePort=11, protocol=tcp, publicPort=11}, {allocatedIP=192.168.60.60, privatePort=22, protocol=tcp, publicPort=22}], instanceID=494}] which is hard to make it to a specific data type